CMakeLists.txt

```diff
@@ -655,7 +655,7 @@ if (BUILD_EXAMPLES AND TARGET s2testing)
   add_subdirectory("doc/examples" examples)
 endif()
 
-if (${SWIG_FOUND} AND ${Python3_FOUND})
+if (SWIG_FOUND AND Python3_FOUND)
```

Comment:
Odd since this hasn't changed in 5 years.

https://github.com/google/s2geometry/blame/473a989af642959b1a52bdb675445ff1a9dd9c09/CMakeLists.txt#L658

What's the exact command that fails for you and what cmake version?

Reply:
It can became an issue since this https://github.com/google/s2geometry/commit/efb124d8eaf3433323d3e877dedd5e94a63339a3

Reply:
Maybe you can try following patch in github CI?
https://github.com/google/s2geometry/compare/master...MBkkt:s2geometry:patch-11

Reply:
The easiest way to test it is to make a PR.

Reply:
I think I find main difference in `if (condition)` https://cmake.org/cmake/help/latest/command/if.html#variable-expansion but still don't understand why `if (${undefined_var} AND ${defined_as_true_var})` doesn't work